winch/codegen/src/isa/aarch64/masm.rs

```diff
@@ -429,8 +430,20 @@ impl Masm for MacroAssembler {
         self.asm.load_constant(0, reg);
     }
 
-    fn popcnt(&mut self, _context: &mut CodeGenContext, _size: OperandSize) {
-        todo!()
+    fn popcnt(&mut self, context: &mut CodeGenContext, size: OperandSize) {
+        let src = context.pop_to_reg(self, None);
+        let tmp = context.reg_for_class(RegClass::Float, self);
+        self.asm.mov_to_fpu(src.into(), tmp, size);
+        self.asm.cnt(tmp);
+        match size {
+            OperandSize::S8 => {}
+            OperandSize::S16 => self.asm.addp_rrr(tmp, tmp, tmp, VectorSize::Size8x8),
```

Comment:
If I'm not wrong, Wasm only defined `{i32, i64}_popcnt` so I believe we only need to handle the `S32 | S64` case and handle all the other operand sizes with `unreachable!()` instead of `unimplemented`, which also means that we can probably drop the `addp_rrr` implementation.

Reply:
You're right, I used `lower.isle` as a reference but CLIF is different from WASM. 

> handle all the other operand sizes with `unreachable!()` instead of `unimplemented`

I just removed the match, please tell me if you think an assertion is needed